pkg/controller/events.go

```diff
@@ -50,7 +50,7 @@ func (c *Controller) sendEventToWebhook(r *flaggerv1.Canary, eventType, template
 }
 
 func (c *Controller) alert(canary *flaggerv1.Canary, message string, metadata bool, severity flaggerv1.AlertSeverity) {
-	if c.notifier == nil && len(canary.GetAnalysis().Alerts) == 0 {
+	if len(canary.GetAnalysis().Alerts) == 0 {
```

Comment:
This breaks the global notifier, I think it's safe to remove this `if` altogether.